amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java

```diff
@@ -214,8 +215,10 @@ public void executeImpl()
 
                 // execute search
                 ResultSet results = searchService.query(params);
-                List<NodeRef> resultNodes = results.getNodeRefs();
                 hasMore = results.hasMore();
+                if(results.length() != 0) {
+                    resultNodes=results.getNodeRefs().stream().filter(node ->!freezeService.isFrozenOrHasFrozenChildren(nodeService.getPrimaryParent(node).getParentRef())).collect(Collectors.toList());
```

Comment:
Put the comments as what is these changes doing.

Reply:
Added